src/backend/src/game/game.controller.ts

```diff
@@ -64,15 +64,19 @@ export class GameController {
     @Body('playerId') playerId: number,
     @Body('howMany') howMany: number,
     @Body('timeTaken') timeTaken: number,
-    @Body('roundId') roundId: number,
+    @Body('roundNumber') roundNumber: number,
   ): Promise<number> {
     const player = await this.playersRepository.findOne(playerId, {
-      relations: ['grabs'],
+      relations: ['grabs', 'game'],
     });
 
-    let round = await Round.findOne(roundId, {
-      relations: ['playerMoves', 'game'],
-    }); // Check for updates
+    let round = await this.roundService.findByRoundNumber(
+      roundNumber,
+      player.game.id,
+    );
+    if (!round) {
+      throw new BadRequestException('Round number does not exist');
+    } // Check for updates
     this.validatePlayerAndRound(player, round);
```

Comment:
I think this is duplicated logic. We can remove the !round check, bc we should be checking it in validate player and round